syft/frameworks/torch/hook/hook_args.py

```diff
@@ -122,7 +122,7 @@ def hook_method_args(attr, method_self, args, kwargs):
         new_self, new_args = hook_args((method_self, args))
 
     except (IndexError, KeyError, AssertionError):  # Update the function in case of an error
-        args_hook_function, _ = build_hook_args_function((method_self, args))
+        args_hook_function, _ = build_hook_for_args_with_function((method_self, args))
```

Comment:
same here maybe unwrap_args_in_function is more precise

Reply:
`unwrap_args_with_function`  maybe more precise.